noria-server/src/controller/inner.rs

```diff
@@ -756,16 +731,29 @@ impl ControllerInner {
             }
         };
 
-        self.find_view_for(node).map(|r| {
-            let domain = self.ingredients[r].domain();
-            let columns = self.ingredients[r].fields().to_vec();
+        // reader should be a child of the given node. however, due to sharding and replication,
+        // it may not be an *immediate* child. furthermore, once we go beyond depth 1, we may
+        // accidentally hit an *unrelated* reader node. to account for this, nodes cache their
+        // readers so we can easily query data.
```

Comment:
I think this comment should now go away, since we no longer have to look past ingress/egress nodes here.